src/metabase/lib/metadata.cljc

```diff
@@ -169,10 +169,13 @@
       (editable-stages? query stages))))
 
 (mu/defn database-supports? :- :boolean
-  "Does `query`s [[database]] support the given `feature`?"
-  [query   :- ::lib.schema/query
-   feature :- :keyword]
-  (-> query
+  "Does `metadata-providerable`'s [[database]] support the given `feature`?
+
+  Minimize the use of this function. Using it is often a code smell. The lib should not normally be concerned with
+  driver features. See https://github.com/metabase/metabase/pull/55206#discussion_r2017378181"
```

Comment:
@bshepherdson I initially added a kondo linter as well, but after turning it on, it didn't feel super helpful. Most of our existing linters seem to be focused on actionable advice like "don't use x, use y instead", or "oops, looks like you're about to check in a call to *obvious debugging function*". When a linter like that runs, you'll think "thank goodness it told me!". My concern is that people will see a linter warning that says "don't use x, and write a comment to justify your usage" as mildly annoying, add a comment and move on. I still have the changes stashed though, and am open to adding them if you feel strongly about the linter.

Reply:
Also, to be fair, part of the problem was that I wasn't sure what to write as justification for some of the existing uses. Like, of course we don't want to return join strategies that aren't supported by the database from `available-join-strategies`, and it just felt awkward to justify that with a comment.

Reply:
I'm content to move ahead without the linter.

The library is intended to be the source of truth, to Know About all the fiddly details so you don't have to. Part of achieving that is to know about the database `:features`, and eg. exclude join strategies, expressions and aggregation operators that the database doesn't support.